src/server/services/collection.service.ts

```diff
@@ -792,3 +807,58 @@ export const updateCollectionItemsStatus = async ({
     // Ignore errors
   }
 };
+
+export const bulkSaveItems = async ({
+  input: { collectionId, articleIds = [], modelIds = [], imageIds = [], postIds = [] },
+  user,
+}: {
+  input: BulkSaveCollectionItemsInput;
+  user: SessionUser;
+}) => {
+  const collection = await dbRead.collection.findUnique({
+    where: { id: collectionId },
+    select: { type: true },
+  });
+  if (!collection) throw throwNotFoundError('No collection with id ' + collectionId);
+
+  let data: Prisma.CollectionItemCreateManyInput[] = [];
+  if (
+    articleIds.length > 0 &&
+    (collection.type === CollectionType.Article || collection.type === null)
+  ) {
+    data = articleIds.map((articleId) => ({
+      articleId,
+      collectionId,
+      addedById: user.id,
+    }));
+  }
+  if (
+    modelIds.length > 0 &&
+    (collection.type === CollectionType.Model || collection.type === null)
+  ) {
+    data = modelIds.map((modelId) => ({
+      modelId,
+      collectionId,
+      addedById: user.id,
+    }));
+  }
+  if (
+    imageIds.length > 0 &&
+    (collection.type === CollectionType.Image || collection.type === null)
+  ) {
+    data = imageIds.map((imageId) => ({
+      imageId,
+      collectionId,
+      addedById: user.id,
+    }));
+  }
+  if (postIds.length > 0 && (collection.type === CollectionType.Post || collection.type === null)) {
+    data = postIds.map((postId) => ({
+      postId,
+      collectionId,
+      addedById: user.id,
+    }));
```

Comment:
if we're considering null types here, we should append to the data, not re-assign. 
Also, be sure to add the relevant status:
```js
status: permission.writeReview
            ? CollectionItemStatus.REVIEW
            : CollectionItemStatus.ACCEPTED,
```